### PR TITLE
[dmlib] Audit XML comments in `Azure.Storage.DataMovement.Blobs` dmlib library 

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR contains the XML comment changes for the `Azure.Storage.DataMovement.Blobs` library.

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs [hunk 1]
```diff
@@ -28,19 +28,19 @@ public class BlobDirectoryStorageResourceContainer : StorageResourceContainer
...
         /// <summary>
-        /// Returns the path
+        /// Gets the path of the storage resource.
         /// </summary>
         public override string Path => _directoryPrefix;
 
         internal Uri _uri;
 
         /// <summary>
-        /// Obtains the Uri of the blob directory resource, which means we can list
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs:43]: Just want to call out if this docstring is accurate?

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs [hunk 2]
```diff
@@ -96,10 +96,10 @@ public override StorageResource GetChildStorageResource(string encodedPath)
         }
 
         /// <summary>
-        /// Retrieves a single blob resoruce based on this respective resource.
+        /// Retrieves a single blob resource based on this respective resource.
         /// </summary>
-        /// <param name="encodedPath"></param>
-        /// <param name="type"></param>
+        /// <param name="encodedPath">A URL-encoded path as it would appear in a request URI.</param>
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs:101]: Want to callout to check if this docstring is accurate (I modeled it after the description of encoded path found in .NET Blobs package)

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs [hunk 3]
```diff
@@ -27,17 +27,17 @@ public class AppendBlobStorageResource : StorageResource
...
 
         /// <summary>
-        /// Gets the path of the resource.
+        /// Gets the path of the storage resource.
         /// </summary>
         public override string Path => _blobClient.Name;
 
         /// <summary>
-        /// Defines whether the object can produce a SAS URL
+        /// Defines whether the storage resource type can produce a SAS URL.
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs:40]: Sorry, wanted to call out this one to make sure this is accurate (i.e. does tell you about SAS URL viability)

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobStorageResourceContainer.cs [hunk 4]
```diff
@@ -14,54 +14,59 @@
...
+        /// Gets the URL of the storage resource. If the URL can be obtained, this storage resource can be listed.
         /// </summary>
         public override Uri Uri => _blobContainerClient.Uri;
 
         /// <summary>
-        /// Creates new blob client based on the parent container client
+        /// Retrieves a single blob resource based on this respective resource.
         /// </summary>
-        /// <param name="path"></param>
+        /// <param name="path">The path to the storage resource.</param>
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobStorageResourceContainer.cs:58]: Want to check if this should be changed to `encodedPath` or if that is inaccurate? Not that there is anything wrong with `path`, but rather that this is the only place where the parameter is named `path`.

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResourceServiceCopyOptions.cs [hunk 5]
```diff
@@ -19,7 +19,7 @@ public class AppendBlobStorageResourceServiceCopyOptions
         public TransferCopyMethod CopyMethod { get; set; }
 
         /// <summary>
-        /// <see cref="AppendBlobRequestConditions"/> to apply when
+        /// See <see cref="AppendBlobRequestConditions"/> to apply when
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResourceServiceCopyOptions.cs:22]: Is this one not optional like the others?

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlockBlobStorageResourceUploadOptions.cs [hunk 6]
```diff
@@ -20,12 +21,12 @@ public class BlockBlobStorageResourceUploadOptions
     {
         /// <summary>
         /// Optional standard HTTP header properties that can be set for the
-        /// new append blob.
+        /// new block blob.
         /// </summary>
         public BlobHttpHeaders HttpHeaders { get; set; }
 
         /// <summary>
-        /// Optional custom metadata to set for this append blob.
```
Comment (vincenttran-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlockBlobStorageResourceUploadOptions.cs:28]: Assuming this was a copy-paste error? Just checking 😄

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs (hunk 3 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs:40]: Nah. I think this was a mistake on my end. It used to be a SAS url, but here we just return only the URL. So we can omit SAS from the comment.

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResourceServiceCopyOptions.cs (hunk 5 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResourceServiceCopyOptions.cs:22]: It's optional.

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs:43]: Hm. I'm not sure what I was trying to write here, but let's just keep it simple as "Gets the URL of the storage resource."

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs:101]: Yeah this is accurate.

Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainer.cs:101]: Nvm, let's just remove URL-encoded. I'm going to rename this back to path in another PR. (see the other comment you made about `path` and `encodedPath`

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs (hunk 3 above)
Comment (vincenttran-msft) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/src/AppendBlobStorageResource.cs:40]: Sounds good. Fixed all occurrences from `SAS URL` to just `URL` in this lib

**Pull request merged**